app/src/main/java/org/theotech/ceaselessandroid/activity/MainActivity.java

```diff
@@ -121,4 +152,33 @@ public void notify(FragmentState fragmentState) {
         // update current fragment
         currentFragment = fragmentState;
     }
+
+    private void populateContacts() {
+        if (ContextCompat.checkSelfPermission(this, Manifest.permission.READ_CONTACTS)
+            != PackageManager.PERMISSION_GRANTED) {
+            // Should we show an explanation?
+            if (ActivityCompat.shouldShowRequestPermissionRationale(
+                        this, Manifest.permission.READ_CONTACTS)) {
+                Log.d(TAG, "Should show rationale");
+                // TODO: Display a fragment or something here.
+                // From the developers handbook:
+                // Show an explanation to the user *asynchronously* -- don't block
+                // this thread waiting for the user's response! After the user
+                // sees the explanation, try again to request the permission.
+                final Handler handler = new Handler();
+                handler.postDelayed(new Runnable() {
+                    @Override
+                    public void run() {
+                        populateContacts();
```

Comment:
Yea, if you open again, it will ask permission again.  If you want better behavior, I think that can wait for a different commit.  This at least runs without crashing.